File: day15/day15.pl
# Day 15 in perl.
# We had some elves play a memory game with us.

open STDIN,'day15.txt'; # omit this if reading from stdin

$/=',';chop   ,$M          {$_}=[$.,$.
]while<>;$x   =$M          {$l=$M{$l}[
0]+           -$M          {$l
}[+           1]}          ||=
[$.,$.]       ,($          .==2020
and$o=$       l),          unshift
($x           ,$.          ),!
pop           $x,          "S"
until$.++==   30000000;"   &";
print"P1: $   o\nP2: $l\   \n"

File: 2020/day15/day15.rb
map = Hash.new { _1[_2] = [$., $.] }

open('day15.txt').each_line(',') { map[_1.to_i] }

last = map.keys.last

map[last = map[last].reduce(&:-)].unshift($.).pop until 30000001 == $. += 1

p last
